irohad/multi_sig_transactions/hash.hpp

```diff
@@ -35,12 +24,7 @@ namespace iroha {
     class PeerHasher {
      public:
       std::size_t operator()(
-          const std::shared_ptr<shared_model::interface::Peer> &obj) const {
-        return hasher(obj->address() + obj->pubkey().hex());
-      }
-
-     private:
-      std::hash<std::string> hasher;
+          const std::shared_ptr<shared_model::interface::Peer> &obj) const;
```

Comment:
Add <memory> & iface::Peer fdecl please

Reply:
https://github.com/hyperledger/iroha/blob/99b8a32a65fb69575f01b9a5950042ecde290754/irohad/multi_sig_transactions/mst_types.hpp#L17
Since `mst_types` is included in hash, maybe there is no need to add another forward declaration?